kubernetes-ingress/tls-with-cert-manager.md

```diff
@@ -0,0 +1,121 @@
+# Ingress example with TLS termination using cert-manager
+
+This example builds on the [HTTP](http.md) and [gRPC](grpc.md) ingress
+examples, adding TLS termination. This example is similar to the
+[TLS](tls.md) ingress example, in which you had to manually create a
+self-signed CA using `minica`. In this example, you will be using
+[cert-manager](https://cert-manager.io/), a Kubernetes-native operator that
+issues and renews TLS certificates for you.
+
+## Create TLS certificate and private key
+
+Let us install cert-manager:
+
+```sh
+helm repo add jetstack https://charts.jetstack.io
+helm install cert-manager jetstack/cert-manager --version v1.7.1 --namespace cert-manager --set installCRDs=true --create-namespace
+```
+
+Now, create a CA Issuer:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/ca-issuer.yaml
+```
+
+## Deploy the ingress
+
+If you previously ran the HTTP and/or gRPC ingress demos, delete the ingresses:
+
+```sh
+kubectl delete ingress basic-ingress grpc-ingress
+```
+
+The ingress configuration for this demo provides the same routing as those demos
+but with the addition of TLS termination. Deploy this ingress:
+
+```sh
+kubectl apply -f https://raw.githubusercontent.com/cilium/cilium-service-mesh-beta/main/kubernetes-ingress/tls-ingress.yaml
+```
+
+This creates a LoadBalancer service, which after around 30 seconds or so should
+be populated with an external IP address.
+
+```console
+$ kubectl get ingress
+NAME          CLASS    HOSTS                                            ADDRESS        PORTS     AGE
+tls-ingress   cilium   hipstershop.cilium.rocks,bookinfo.cilium.rocks   35.195.24.75   80, 443   6m5s
+```
+
+To tell cert-manager that this Ingress needs a certificate, annotate the
+Ingress with the name of the CA issuer we previously created:
+
+```sh
+kubectl annotate ingress tls-ingress cert-manager.io/issuer=ca-issuer
+```
+
+This creates a Certificate object along with a Secret containing the TLS
+certificate.
+
+```console
+$ kubectl get certificate,secret demo-cert
+NAME                                    READY   SECRET      AGE
+certificate.cert-manager.io/demo-cert   True    demo-cert   33m
+
+NAME               TYPE                DATA   AGE
+secret/demo-cert   kubernetes.io/tls   3      33m
+```
+
```

Comment:
```suggestion
_Note: if you previously followed the the basic [TLS example](TLS.md) there was a pre-existing secret called `demo-cert`. This will be properly "updated" by cert-manager but the updated secret won't be picked up by Cilium ([#22](https://github.com/cilium/cilium-service-mesh-beta/issues/22)). You can work around this by deleting and re-creating the service `cilium-ingress-tls-ingress` after checking that cert-manager has properly created the secret. One way to do this is to delete and re-create the ingress config file tls-ingress.yaml_ 
```

Reply:
Done in 06045ec210bbea3246355602364105b6df3e3e0a, thank you!!